File: data/readable_format/cpp/file166.cpp
#include <iostream>
#include <cmath>

// Global variables
bool use_gpu = false, noncolin = false, lspinorb = false, tqr = false, spline_ps = false;
int nat = 0, nspin = 1, nhm = 0, nsp = 0, nbetam = 0, nwfcm = 0, nqxq, nqx;
double cell_factor = 1.2, ecutrho = 20.0, qnorm = 1.0, dq = 0.1, ecutwfc = 15.0;
double lmaxq = 0.0, lmaxkb = 0.0, nh = 0.0;

void allocate_nlpot() {
    if (spline_ps && cell_factor <= 1.1) {
        cell_factor = 1.1;
    }
    
    nqxq = static_cast<int>(((std::sqrt(ecutrho) + qnorm) / dq + 4) * cell_factor);
    nqx = static_cast<int>((std::sqrt(ecutwfc) / dq + 4) * cell_factor);
}

int main() {
    allocate_nlpot();
    std::cout << "C++ Stdout: nqxq = " << nqxq << std::endl;
    std::cout << "C++ Stdout: nqx = " << nqx << std::endl;
    return 0;
}

File: data/readable_format/fortran/file166.f90
MODULE global_vars
  IMPLICIT NONE
  LOGICAL :: use_gpu = .FALSE., noncolin = .FALSE., lspinorb = .FALSE., tqr = .FALSE., spline_ps = .FALSE.
  INTEGER :: nat = 0, nspin = 1, nhm = 0, nsp = 0, nbetam = 0, nwfcm = 0, nqxq, nqx
  REAL :: cell_factor = 1.2, ecutrho = 20.0, qnorm = 1.0, dq = 0.1, ecutwfc = 15.0
  REAL :: lmaxq = 0.0, lmaxkb = 0.0, nh = 0.0
END MODULE global_vars

SUBROUTINE allocate_nlpot
  USE global_vars
  IF (spline_ps .AND. cell_factor <= 1.1) cell_factor = 1.1
  nqxq = INT(((SQRT(ecutrho) + qnorm) / dq + 4) * cell_factor)
  nqx = INT((SQRT(ecutwfc) / dq + 4) * cell_factor)
END SUBROUTINE allocate_nlpot

PROGRAM test_allocate_nlpot
  USE global_vars
  CALL allocate_nlpot
  PRINT *, "Fortran Stdout: nqxq =", nqxq
  PRINT *, "Fortran Stdout: nqx =", nqx
END PROGRAM test_allocate_nlpot